server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -49,6 +49,27 @@ public class SegmentReplicationTargetService implements IndexEventListener {
 
     private final SegmentReplicationSourceFactory sourceFactory;
 
+    // Empty Implementation, only required while Segment Replication is under feature flag.
+    public static final SegmentReplicationTargetService NO_OP = new SegmentReplicationTargetService() {
+        @Override
+        public void beforeIndexShardClosed(ShardId shardId, IndexShard indexShard, Settings indexSettings) {
+            // NoOp;
+        }
+
+        @Override
+        public synchronized void onNewCheckpoint(ReplicationCheckpoint receivedCheckpoint, IndexShard replicaShard) {
+            // noOp;
+        }
+    };
+
+    // Used only for empty implementation.
+    private SegmentReplicationTargetService() {
+        threadPool = null;
+        recoverySettings = null;
+        onGoingReplications = null;
+        sourceFactory = null;
+    }
```

Comment:
Maybe we don't have to explicitly set them to null a private constructor would by default have them null implicitly
```
private SegmentReplicationTargetService() {
}
```

Reply:
I'd rather keep these as final fields, which will require the explicit initialization to null here.